sparse/_compressed/common.py

```diff
@@ -41,6 +41,9 @@ def concatenate(arrays, axis=0, compressed_axes=None):
     data = np.concatenate([arr.data for arr in arrays])
     ptr_len = arrays[0].indptr.shape[0]
     nnz = arrays[0].nnz
+    total_nnz = sum([arr.nnz for arr in arrays[:-1]])
```

Comment:
```suggestion
    total_nnz = sum(int(arr.nnz) for arr in arrays[:-1])
```

To make sure we avoid overflows, and don't construct a list.